src/report.sh

```diff
@@ -321,101 +333,100 @@ function report_parse()
-        fi
-      elif [[ "$week" == 1 ]]; then
-        # First day of the week
-        week=0
-        if [[ -n "$option" ]]; then
-          options_values['WEEK']=$(get_week_beginning_day "$option")
-          if [[ "$?" != 0 ]]; then
-            complain "Invalid parameter: $option"
-            return 22 # EINVAL
-          fi
-        fi
-      elif [[ "$month" == 1 ]]; then
-        month=0
-        if [[ -n "$option" ]]; then
-          # First day of the month
-          options_values['MONTH']=$(date_to_format "$option/01" '+%Y/%m')
-          if [[ "$?" != 0 ]]; then
-            complain "Invalid parameter: $option"
-            return 22 # EINVAL
-          fi
-        fi
-      elif [[ "$year" == 1 ]]; then
-        year=0
-        if [[ -n "$option" ]]; then
-          options_values['YEAR']=$(date_to_format "$option/01/01" +%Y)
-          if [[ "$?" != 0 ]]; then
-            complain "Invalid parameter: $option"
-            return 22 # EINVAL
-          fi
-        fi
-      elif [[ "$output" == 1 ]]; then
-        options_values['OUTPUT']="$option"
-      fi
-    fi
+  eval "set -- $options"
+
+  while [[ "$#" -gt 0 ]]; do
+    case "$1" in
+      --day)
+        options_values['DAY']=$(get_today_info '+%Y/%m/%d')
+        reference='DAY'
+        reference_count+=1
+        shift
+        ;;
+      --week)
+        options_values['WEEK']=$(get_week_beginning_day)
+        reference='WEEK'
+        reference_count+=1
+        shift
+        ;;
+      --month)
+        options_values['MONTH']=$(get_today_info '+%Y/%m')
+        reference='MONTH'
+        reference_count+=1
+        shift
+        ;;
+      --year)
+        options_values['YEAR']=$(date +%Y)
+        reference='YEAR'
+        reference_count+=1
+        shift
+        ;;
+      --output)
+        options_values['OUTPUT']=""
+        output=1
+        shift
+        ;;
+      --statistics)
+        is_statistics=1
+        shift
+        ;;
+      --)
+        shift
+        break
+        ;;
+      *)
+        options_values['ERROR']="Unrecognized argument: $1"
+        return 22 # EINVAL
+        shift
+        ;;
+    esac
   done
 
-  if [[ "$reference" -gt 1 ]]; then
+  if [[ "$reference_count" -gt 1 ]]; then
     complain 'Please, only use a single time reference'
     return 22
-  elif [[ "$reference" == 0 ]]; then
-    # If no option, set day as a default
+  fi
+
+  if [[ "$output" == 1 ]]; then
+    if [[ "$is_statistics" == 1 ]]; then
+      options_values['ERROR']="--output cannot be used with --statistics."
+      return 22 # EINVAL
+    fi
+
+    if [[ "$#" == 1 ]]; then
+      options_values['OUTPUT']="$1"
+      shift
+    elif [[ "$#" == 2 ]]; then
+      options_values['OUTPUT']="$2"
+    else
+      options_values['ERROR']="Invalid path for the --output argument."
+      return 22 # EINVAL
+    fi
+  fi
+
+  if [[ "$#" == 0 && "$reference" == 'DAY' ]]; then
     options_values['DAY']=$(get_today_info '+%Y/%m/%d')
+  elif [[ -n $1 ]]; then
```

Comment:
Why are you checking `$1`? At this point, I expected that `$@` is empty; if not, something might be wrong.